classes/plugins/PluginRegistry.inc.php

```diff
@@ -89,21 +89,12 @@ public static function register($category, $plugin, $path, $mainContextId = null
         $pluginName = $plugin->getName();
         $plugins = & self::getPlugins();
 
-        // Allow the plugin to register.
-        if (!$plugin->register($category, $path, $mainContextId)) {
+        // If the plugin is already loaded or failed/refused to register
+        if (isset($plugins[$category][$pluginName]) || !$plugin->register($category, $path, $mainContextId)) {
```

Comment:
I arrived here after the `StaticPagesPlugin` tried to run its migration twice. I checked the hooks and there were many duplicated entries registered.

Another potencial side-fix would be to block the `HookRegistry` from registering the same hook+callback, but as a closure can be passed, I decided to not bother with it for now.